### PR TITLE
feat: 🍰 Switch User Role As Admin

### LINKED ISSUE #4124: 🚀 [Feature] Admin Can Give Roles To Users
## :rocket: Feature

An admin should be able to give rolls to and change roles of users.

### User Problem

At the moment no user, not even an admin, can change rolls of users.
We do it per direct database manipulation.

### Implementation

I suggest to have a popup menu for the role of a user in the admins users list.

Bildschirmfoto 2021-01-11 um 09 04 02

### Design & Layout

See above ☝🏼

### Validation

None

### Additional context

None

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest
An admin can change the role a user has

### Issues
- fixes #4124

### Todo

- [ ] Do we have to ensure that there is at least one admin remaining? What happens if the only admin changes their role to user?

### CODE REVIEW

**Comment by Mogge** [comment]
Can an admin change her own role? This way we could avoid easily that at least one admin is remaining.

**Comment by Mogge** [comment]
@tippanamadhusudan I added the tests to the backend. I also made sure, that the current user cannot change the own role to make sure that there is always at least one admin left. May be you can catch this in the frontend as well. The backend throws an error when current user tries to change the own role.

File: webapp/graphql/admin/Roles.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      enumValues {
+        name
+      }
+    }
+  }
+`
+
+export const updateUserRole = (role, id) => {
+  return gql`
+        mutation {
```
[the remark this answers is not in the document]
Comment (Tirokk) [reply webapp/graphql/admin/Roles.js:16]: If you could solve this by following @Mogge suggestion … @tippanamadhusudan

File: webapp/pages/admin/users.vue [hunk 2]
```diff
@@ -48,6 +48,23 @@
         <template #createdAt="scope">
           {{ scope.row.createdAt | dateTime }}
         </template>
+
+        <template slot="role" slot-scope="scope">
+          <ApolloQuery :query="FetchAllRoles">
+            <template v-slot="{ result: { data } }">
+              <template v-if="data">
+                <select
```
[the remark this answers is not in the document]
Comment (Tirokk) [reply webapp/pages/admin/users.vue:56]: … and try this as well @tippanamadhusudan

File: webapp/pages/admin/users.vue [hunk 3]
```diff
@@ -174,6 +193,16 @@ export default {
         }
       }
     },
+    changeUserRole(id, event) {
+      const newRole = event.target.value
+      // this.role = '';
+      // const id = this.User[0].id;
+      this.$apollo
+        .mutate({
```
[the remark this answers is not in the document]
Comment (Tirokk) [reply webapp/pages/admin/users.vue:203]: … and this. @tippanamadhusudan

File: webapp/pages/admin/users.vue [hunk 4]
```diff
@@ -174,6 +193,16 @@ export default {
...
+          mutation: updateUserRole(newRole, id),
+        })
```
[the remark this answers is not in the document]
Comment (Tirokk) [reply webapp/pages/admin/users.vue:209]: I removed the `console.log` already. Hope this is okay … !?
Was a linting error …

**Review by Tirokk — CHANGES_REQUESTED** [review]
Hey @tippanamadhusudan ,
your first PR is close to get merged into the master !!! 😍

And it’s a really important one, because we need it urgent for the first network `wir.social` going online at the beginning of next week! 🚀🚀💫

If you could follow @Mogge suggestions 👇🏼 you have it quickly ready. 🙏🏼💪🏼

Thank you very much for helping !!! 🍀

**Comment by tippanamadhusudan** [comment]
Ok, I will make changes according to @mogge.

**Review by Tirokk — APPROVED** [review]
You both got it together !!!

A big thank to @tippanamadhusudan !!! This will go into our first live version of ocelot.social: `wir.social` next week.
Congratulations !!! 🚀🚀💫💫
… and to @Mogge as well for sure! 😘

Just a little hint … 👇🏼

Gorzilla

**Pull request merged**